### PR TITLE
Tests for CASSANDRA-9474

### PR DESCRIPTION
**Pull request by pauloricardomg**

### CODE REVIEW

File: replication_test.py [hunk 1]
```diff
@@ -488,3 +488,35 @@ def _test_rf_on_snitch_update(self, nodes, rf, snitch_class_name, snitch_config_
 
         # nodes have joined racks, check endpoint counts again
         self.check_endpoint_count('testing', 'rf_test', cluster.nodelist(), rf)
+
+    @since('2.1')
```
Comment (ptnapoleon) [inline replication_test.py]: No need for this.

File: replication_test.py [hunk 2]
```diff
@@ -488,3 +488,35 @@ def _test_rf_on_snitch_update(self, nodes, rf, snitch_class_name, snitch_config_
...
+    def test_switch_data_center_startup_fails(self):
+        """
+        @jira_ticket CASSANDRA-9474
+
+        Confirm that switching data centers fails to bring up the node.
+        """
+        expected_error = r'Cannot start node if snitch\'s data center (.*) differs from previous data center (.*)\.' \
```
Comment (ptnapoleon) [inline replication_test.py]: You can wrap this string in `"` and then you don't have to escape the `'` characters.

**Comment by pauloricardomg** [comment]
Fixed nits

**Comment by ptnapoleon** [comment]
+1

**Comment by ptnapoleon** [comment]
Moved this to #694. Sorry for merging early.

**Pull request merged**